sdk/servicebus/Azure.Messaging.ServiceBus/tests/Client/ServiceBusClientTests.cs

```diff
@@ -406,6 +406,37 @@ public void ValidateClientProperties()
             Assert.IsFalse(client.IsClosed);
         }
 
+        [Test]
+        [TestCase("queue1", "Endpoint=sb://not-real.servicebus.windows.net/;SharedAccessKeyName=DummyKey;SharedAccessKey=[not_real];EntityPath=queue1")]
+        [TestCase("queue1", "Endpoint=sb://not-real.servicebus.windows.net/;SharedAccessKeyName=DummyKey;SharedAccessKey=[not_real];")]
+        [TestCase("topic1", "Endpoint=sb://not-real.servicebus.windows.net/;SharedAccessKeyName=DummyKey;SharedAccessKey=[not_real];")]
+        [TestCase("topic1/Subscriptions/sub1", "Endpoint=sb://not-real.servicebus.windows.net/;SharedAccessKeyName=DummyKey;SharedAccessKey=[not_real];")]
+        [TestCase("topic1/Subscriptions/sub1", "Endpoint=sb://not-real.servicebus.windows.net/;SharedAccessKeyName=DummyKey;SharedAccessKey=[not_real];EntityPath=topic1")]
+        [TestCase("topic1/Subscriptions/sub1", "Endpoint=sb://not-real.servicebus.windows.net/;SharedAccessKeyName=DummyKey;SharedAccessKey=[not_real];EntityPath=topic1/Subscriptions/sub1")]
+        public void ValidateEntityNameAllowsValidPaths(string entityName, string connectionString)
+        {
+            var client = new ServiceBusClient(connectionString);
+            client.ValidateEntityName(entityName);
```

Comment:
```suggestion
            Assert.That(
                () => client.ValidateEntityName(entityName), 
                Throws.Nothing);
```
Functionally the same, but I believe this better conveys the intent to validate there isn't an exception.   Without the assert, I always ask myself if we forgot to finish implementing.